gui/wxpython/gui_core/infobar.py

```diff
@@ -0,0 +1,89 @@
+
+import wx
+import wx.aui
+try:
+    import wx.lib.agw.infobar as IB
+except ImportError:
+    import wx.lib.infobar as IB
+
+
+class InfoBar(IB.InfoBar):
+    """
+    Custom Info bar
+    """
+    def __init__(self, parent):
+        IB.InfoBar.__init__(self, parent)
+
+        colBg = wx.SystemSettings.GetColour(wx.SYS_COLOUR_HIGHLIGHT)
+        self.SetBackgroundColour(colBg)
+        self.SetOwnForegroundColour(wx.SystemSettings.GetColour(wx.SYS_COLOUR_HIGHLIGHTTEXT))
+        self._text.SetBackgroundColour(colBg)
+        self._button.SetBackgroundColour(colBg)
+
+        # LAYOUT
+        sizer = wx.BoxSizer(wx.VERTICAL)
+        self.subSizerText = wx.BoxSizer(wx.HORIZONTAL)
+        self.subSizerButtons = wx.BoxSizer(wx.HORIZONTAL)
+        self.subSizerText.Add(self._icon, wx.SizerFlags().Centre().Border())
+        self.subSizerText.Add(self._text, 200, wx.ALIGN_CENTER_VERTICAL)
+        self.subSizerText.AddStretchSpacer()
+        self.subSizerText.Add(self._button, wx.SizerFlags().Centre().Border())
+        self.subSizerButtons.AddStretchSpacer()
+        sizer.Add(self.subSizerText, wx.SizerFlags().Expand())
+        sizer.Add(self.subSizerButtons, wx.SizerFlags().Expand())
+        self.SetSizer(sizer)
+
+    def AddButton(self, btnid, label, bitmap=wx.NullBitmap):
+        """
+        Adds a button to be shown in the info bar.
+        """
+        sizer = self.GetSizer()
+
+        assert sizer is not None, "Sizer must be created first"
+
+        # user-added buttons replace the standard close button so remove it if we
+        # hadn't done it yet
+        if sizer.Detach(self._button):
+            self._button.Hide()
+
+        if bitmap.IsOk():
+            # Add the bitmap to the button
+            button = wx.BitmapButton(self, id=btnid, bitmap=bitmap, name="Create new Location")
+            button.SetBitmap(bitmap, wx.LEFT)
+            button.SetBitmapMargins((2, 2))  # default is 4 but that seems too big to me.
```

Comment:
Let's leave out the bitmap from this PR. I think more discussion is need on what do we want from this. If there is only the icon, what would be the advantage, esp. when right above that we have the same image and in both cases, user would have to take additional step to find out what it is.

Getting the general implementation right might be also challenging given the current code for label/name. So, let's solve this in a separate PR.